integration_tests/test_str_01.py

```diff
@@ -59,6 +61,13 @@ def test_str_join2():
     res:str = a.join(p)
     assert res == "a**b"
 
+def test_str_join3():
+    a: str
+    a = "**-"
+    p:list[str] = ["a","b"]
+    res:str = a.join(p)
+    assert res == "a**-b"
```

Comment:
We should probably add a new test for corner cases like when `a` here is an empty `string` or when `p` here is an empty `list`.

Reply:
This can be tackled in a separate/new PR.

Reply:
I have added the test in the same PR.